src/status_im/models/transactions.cljs

```diff
@@ -1,116 +1,440 @@
+             (map (juxt :hash identity)))
+            result))))
+
+(defn- etherscan-transactions [chain account on-success on-error]
+  (if (etherscan-supported? chain)
+    (let [url (get-transaction-url chain account)]
+      (log/debug "HTTP GET" url)
+      (http/get url
+                #(on-success (format-transactions-response % account))
+                on-error))
+    (log/info "Etherscan not supported for " chain)))
+
+(defn- get-transactions [{:keys [web3 chain chain-tokens account-address success-fn error-fn]}]
+  (log/debug "Syncing transactions data..")
+  (etherscan-transactions chain
+                          account-address
+                          success-fn
+                          error-fn)
+  (doseq [direction [:inbound :outbound]]
+    (get-token-transactions web3
+                            chain-tokens
+                            direction
+                            account-address
+                            success-fn)))
+
+;; ---------------------------------------------------------------------------
+;; Periodic background job
+;; ---------------------------------------------------------------------------
+
+(defn- async-periodic-run!
+  ([async-periodic-chan]
+   (async-periodic-run! async-periodic-chan true))
+  ([async-periodic-chan worker-fn]
+   (async/put! async-periodic-chan worker-fn)))
+
+(defn- async-periodic-stop! [async-periodic-chan]
+  (async/close! async-periodic-chan))
+
+(defn- async-periodic-exec
+  "Periodically execute an function.
+  Takes a work-fn of one argument `finished-fn -> any` this function
+  is passed a finished-fn that must be called to signal that the work
+  being performed in the work-fn is finished.
+
+  The work-fn can be forced to run immediately "
+  [work-fn interval-ms timeout-ms]
+  {:pre [(fn? work-fn) (integer? interval-ms) (integer? timeout-ms)]}
+  (let [do-now-chan (async/chan (async/sliding-buffer 1))]
+    (go-loop []
+      (let [timeout (async-util/timeout interval-ms)
+            finished-chan (async/promise-chan)
+            [v ch] (async/alts! [do-now-chan timeout])
+            worker (if (and (= ch do-now-chan) (fn? v))
+                     v work-fn)]
+        (when-not (and (= ch do-now-chan) (nil? v))
+          (try
+            (worker #(async/put! finished-chan true))
+            ;; if an error occurs in work-fn log it and consider it done
+            (catch :default e
+              (log/error "failed to run transaction sync" e)
+              (async/put! finished-chan true)))
+          ;; sanity timeout for work-fn
+          (async/alts! [finished-chan (async-util/timeout timeout-ms)])
+          (recur))))
+    do-now-chan))
+
+;; -----------------------------------------------------------------------------
+;; Helpers functions that help determine if a background sync should execute
+;; -----------------------------------------------------------------------------
+
+(defn- keyed-memoize
```

Comment:
maybe put this one to `status-im.utils.core` it looks generally useful.

Reply:
Yes this makes sense, I was waiting until the next round of work on transactions, to see if its actually needed.